Add yaml-imenu, which enhances the imenu support in yaml-mode

knu opened the pull request:
### Brief summary of what the package does

yaml-imenu enhances the imenu support in yaml-mode.

### Direct link to the package repository

https://github.com/knu/yaml-imenu.el

### Your association with the package

I'm the author of the package.

### Relevant communications with the upstream package maintainer

None

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

purcell commented:
Thanks. A couple of notes:

- This [auto-enabling of the package](https://github.com/knu/yaml-imenu.el/blob/b1d36aa4d1abf37c9874f8239172dc8980b34c40/yaml-imenu.el#L110-L113) is inappropriate. It should be possible to install this package but not activate it, just like the packages in the Emacs standard library. Instead provide an autoloaded `yaml-imenu-enable` which users can call from their init file, or add to their `yaml-mode-hook`.
- If a yaml file contains malicious content, is it still safe to open that file using this code? You should make this clear.

knu commented:
> - This [auto-enabling of the package](https://github.com/knu/yaml-imenu.el/blob/b1d36aa4d1abf37c9874f8239172dc8980b34c40/yaml-imenu.el#L110-L113) is inappropriate. It should be possible to install this package but not activate it, just like the packages in the Emacs standard library. Instead provide an autoloaded `yaml-imenu-enable` which users can call from their init file, or add to their `yaml-mode-hook`.
OK.

> - Don't use `"*.el"` in the recipe: please use `:defaults` instead.
Thanks, that's nice to know.

> - If a yaml file contains malicious content, is it still safe to open that file using this code? You should make this clear.
The parser just parses a document without evaluating it, so it should be safe.

Thanks for your review, I've addressed the above points.

purcell commented:
Great, thanks - merged now. :tada:
